src/dune_rules/pkg_rules.ml

```diff
@@ -1133,18 +1138,24 @@ module DB = struct
   type t =
     { all : Lock_dir.Pkg.t Package.Name.Map.t
     ; system_provided : Package.Name.Set.t
+    ; components : string list
```

Comment:
This change seems wrong. Why should the database include this stuff?

Reply:
Because we have 2 different database one for a dev-tool pkg and the other for a project.

Reply:
Regardless of how many databases are present, no database should have a "components" field. More appropriate would be to lookup the correct database by giving more input to "resolve".